ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisher.java

```diff
@@ -34,33 +35,48 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
 import java.util.stream.Collectors;
+import org.apache.kafka.common.utils.Time;
 
 class PullQueryPublisher implements Flow.Publisher<Collection<StreamedRow>> {
 
   private final ServiceContext serviceContext;
   private final ConfiguredStatement<Query> query;
   private final PullQueryExecutor pullQueryExecutor;
+  private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
   private final long startTimeNanos;
 
   @VisibleForTesting
   PullQueryPublisher(
       final ServiceContext serviceContext,
       final ConfiguredStatement<Query> query,
       final PullQueryExecutor pullQueryExecutor,
+      final Optional<PullQueryExecutorMetrics> pullQueryMetrics,
       final long startTimeNanos
   ) {
     this.serviceContext = requireNonNull(serviceContext, "serviceContext");
     this.query = requireNonNull(query, "query");
     this.pullQueryExecutor = requireNonNull(pullQueryExecutor, "pullQueryExecutor");
+    this.pullQueryMetrics = pullQueryMetrics;
     this.startTimeNanos = startTimeNanos;
   }
 
   @Override
   public synchronized void subscribe(final Subscriber<Collection<StreamedRow>> subscriber) {
     final PullQuerySubscription subscription = new PullQuerySubscription(
         subscriber,
-        () -> pullQueryExecutor.execute(query, serviceContext, Optional.of(false), startTimeNanos)
+        () -> {
+          final PullQueryResult result = pullQueryExecutor.execute(
+              query, serviceContext, Optional.of(false), pullQueryMetrics);
+          if (pullQueryMetrics.isPresent()) {
+            //Record latency at microsecond scale
+            final long nowNanos = Time.SYSTEM.nanoseconds();
+            final double latency = TimeUnit.NANOSECONDS.toMicros(nowNanos - startTimeNanos);
+            pullQueryMetrics.get().recordLatency(latency);
```

Comment:
I am not measuring the request/response here since I don't have access to the HttpServerRequest/HttpServerResponse objects here